R/parameters.R

```diff
@@ -207,96 +249,60 @@
 #' change; default = 0
+#' * parasite - Plasmodium species; default = "falciparum"
 #' * mosquito_limit - the maximum number of mosquitoes to allow for in the
 #' simulation; default = 1.00E+05
 #' * individual_mosquitoes - boolean whether adult mosquitoes are modeled
 #' individually or compartmentally; default = TRUE
+#' * enable_heterogeneity - boolean whether to include heterogeneity in biting
+#' rates; default = TRUE
 #' * r_tol - the relative tolerance for the ode solver; default = 1e-4
 #' * a_tol - the absolute tolerance for the ode solver; default = 1e-4
 #' * ode_max_steps - the max number of steps for the solver; default = 1e6
-#' * enable_heterogeneity - boolean whether to include heterogeneity in biting
-#' rates; default = TRUE
+#' *  progress_bar - display progress bar; default = FALSE
 #'
 #' @export
-get_parameters <- function(overrides = list()) {
-  parameters <- list(
-    dd    = 5,
-    dt    = 5,
-    da    = 195,
-    du    = 110,
-    del   = 6.64,
-    dl    = 3.72,
-    dpl   = .643,
-    mup   = .249,
-    mum   = .1253333,
-    sigma_squared   = 1.67,
-    n_heterogeneity_groups = 5,
-    # immunity decay rates
-    rm    = 67.6952,
-    rvm   = 76.8365,
-    rb    = 10 * 365,
-    rc    = 30 * 365,
-    rva   = 30 * 365,
-    rid   = 10 * 365,
-    # blood immunity parameters
-    b0    = 0.59,
-    b1    = 0.5,
-    ib0   = 43.9,
-    kb    = 2.16,
-    # immunity boost grace periods
-    ub    = 7.2,
-    uc    = 6.06,
-    uv    = 11.4321,
-    ud    = 9.44512,
-    # infectivity towards mosquitos
-    cd    = 0.068,
-    gamma1= 1.82425,
-    cu    = 0.0062,
-    ct    = 0.021896,
-    # unique biting rate
-    a0    = 8 * 365,
-    rho   = .85,
-    # clinical immunity parameters
-    phi0 = .792,
-    phi1 = .00074,
-    ic0   = 18.02366,
-    kc    = 2.36949,
-    # severe disease immunity parameters
-    theta0  = .0749886,
-    theta1  = .0001191,
-    kv      = 2.00048,
-    fv0     = 0.141195,
-    av      = 2493.41,
-    gammav  = 2.91282,
-    iv0     = 1.09629,
-    # delay for infection
-    de        = 12,
-    delay_gam = 12.5,
-    dem       = 10,
-    # asymptomatic immunity parameters
-    fd0   = 0.007055,
-    ad    = 21.9 * 365,
-    gammad= 4.8183,
-    d1    = 0.160527,
-    id0   = 1.577533,
-    kd    = .476614,
-    # mortality parameters
+get_parameters <- function(overrides = list(), parasite = "falciparum") {
+  
+  ## Parasite-specific parameters set in parasite_parameters.csv
+  # fixed state transitions
+  # immunity decay rates
+  # blood immunity parameters
+  # clinical immunity parameters
+  # severe disease immunity parameters
+  # asymptomatic immunity parameters
+  # immunity boost grace periods
+  # infectivity towards mosquitos
+  # unique biting rate
+  # delay for infection
+  # initial immunities
+  
+  parameters <- as.list(
+    parasite_parameters[parasite_parameters$parasite == parasite, !is.na(parasite_parameters[parasite_parameters$parasite == parasite,])])
```

Comment:
You could pre-do this when preparing the data in data-raw/parasite_parameters.R
In the above script why not create something like:
```
parasites <- c("falciparum", "vivax")
parasite_parameters <-  lapply(parasites, function(parasite){
  as.list(
    parasite_parameters[parasite_parameters$parasite == parasite, !is.na(parasite_parameters[parasite_parameters$parasite == parasite,])]
    )
})
names(parasite_parameters) <- parasites
```
and save that as the data object. Then when called in `get_parameters()` you could just retrieve it with `parasite_parameters[[parasite]]`?

Reply:
Just added this in the commit I'm about to push.